jetty-server/src/main/java/org/eclipse/jetty/server/session/FileSessionDataStore.java

```diff
@@ -206,21 +208,24 @@ public void sweepDisk()
         long now = System.currentTimeMillis();
         if (LOG.isDebugEnabled())
             LOG.debug("Sweeping {} for old session files", _storeDir);
-        try
+        try (Stream<Path> stream = Files.walk(_storeDir.toPath(), 1, FileVisitOption.FOLLOW_LINKS))
         {
-            Files.walk(_storeDir.toPath(), 1, FileVisitOption.FOLLOW_LINKS)
+            stream
                 .filter(p -> !Files.isDirectory(p)).filter(p -> !isOurContextSessionFilename(p.getFileName().toString()))
                 .filter(p -> isSessionFilename(p.getFileName().toString()))
+                .collect(Collectors.toList()) // Don't delete whilst walking
                 .forEach(p ->
                 {
-
                     try
                     {
-                        sweepFile(now, p);
+                        if (!Files.deleteIfExists(p))
+                            LOG.warn("Could not delete {}", p.getFileName());
```

Comment:
In general, I'm not sure we need to do the delete outside of the stream: there are plenty of examples if you do a google search where the recommended way to do a recursive directory delete is to use Files.walk/list and delete each file. I can't find any counter examples that says that is a bad idea.

Reply:
OK I have switched back.